src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -299,16 +299,30 @@ public void categorize(
     }
 
     private Node mergeNodes(Node source, Node target, boolean delete) {
-        copyRelationships(source, copyProperties(source, copyLabels(source, target)), delete);
-        if (delete) source.delete();
+        Map<String, Object> propTemp = new HashMap<>();
```

Comment:
can we find a better name? why not just Map<String, Object> properties = source.getAllProperties()

Didn't we say that we just store/keep the existing properties and then set the new ones after deletion of the source node? Not sure why we have to remove the properties individually?